fearless/Common/DataProvider/Sources/Rewards/SubqueryRewardsSource.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import RobinHood
+import FearlessUtils
+import BigInt
+
+final class SubqueryRewardsSource {
+    typealias Model = [SubqueryRewardItemData]
+
+    let address: String
+    let url: URL
+
+    init(
+        address: AccountAddress,
+        url: URL
+    ) {
+        self.address = address
+        self.url = url
+    }
+}
+
+extension SubqueryRewardsSource: SingleValueProviderSourceProtocol {
+    func fetchOperation() -> CompoundOperationWrapper<[SubqueryRewardItemData]?> {
```

Comment:
I think we should use pagination here. So consider to use `DataProviderSourceProtocol` and DataProvider instead of SingleValueProvider